include/simgrid/s4u/mutex.hpp

```diff
@@ -0,0 +1,47 @@
+/* Copyright (c) 2006-2015. The SimGrid Team. All rights reserved.          */
+
+/* This program is free software; you can redistribute it and/or modify it
+ * under the terms of the license (GNU LGPL) which comes with this package. */
+
+#ifndef SIMGRID_S4U_MUTEX_HPP
+#define SIMGRID_S4U_MUTEX_HPP
+
+#include <xbt/base.h>
+#include <simgrid/s4u/forward.hpp>
+#include "src/simix/smx_process_private.h"
+#include "src/simix/smx_private.h"
+
+
+
+namespace simgrid {
+namespace s4u {
+
+/** @brief Mailboxes
+ *
+ * Rendez-vous point for network communications, similar to URLs on which you could post and retrieve data.
+ * They are not network locations: you can post and retrieve on a given mailbox from anywhere on the network.
+ * You can access any mailbox without any latency. The network delay are only related to the location of the
+ * sender and receiver.
+ */
+XBT_PUBLIC_CLASS Mutex {
```

Comment:
If `Mutex` implements unique onwership of the underlying mutex it should not be copyable but moveable.